File: LeetCode/744/744.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        letters.erase(unique(letters.begin(),letters.end()), letters.end());
        for(auto iter = letters.begin(); iter != letters.end(); ++iter)
        {
            if(target < *iter) return *iter;
        }
        return letters[0];
    }
};

File: LeetCode/744/744.rs
impl Solution {
    pub fn next_greatest_letter(letters: Vec<char>, target: char) -> char {
        let first = letters[0];
        for ch in letters.into_iter()
        {
            if(ch as u8 - b'a' > target as u8 - b'a') {return ch;}
        }
        first
        
    }
}
